File: src/main/java/ru/markn/webchat/repository/RoleRepository.java
package ru.markn.webchat.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import ru.markn.webchat.models.Role;

@Repository
public interface RoleRepository extends JpaRepository<Role, Long>
{
    Role findByName(String name);
}

File: src/main/kotlin/ru/markn/webchat/repositories/RoleRepository.kt
package ru.markn.webchat.repositories

import org.springframework.data.jpa.repository.JpaRepository
import ru.markn.webchat.models.Role
import java.util.*

interface RoleRepository : JpaRepository<Role, Long> {
    fun getByName(name: String): Role
    fun findByName(name: String): Optional<Role>
}